deltacat/compute/compactor_v2/utils/exception_handler.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+from pyarrow.lib import ArrowException, ArrowInvalid
+from ray.exceptions import WorkerCrashedError, NodeDiedError, OutOfMemoryError, RayError
+from deltacat.exceptions import (
+    UploadTableThrottlingError,
+    DependencyRayWorkerDiedError,
+    DependencyRayError,
+    DependencyBotocoreError,
+    DependencyRayOutOfMemoryError,
+    DependencyPyarrowError,
+)
+from botocore.exceptions import ClientError as BotocoreClientError
+from deltacat.aws.constants import UPLOAD_TABLE_POSSIBLE_ERROR_CODE
+
+
+def handle_compaction_step_exception(e: Exception, task_id: Optional[str] = None):
+    if isinstance(e, RayError):
+        _handle_dependency_ray_error(e, task_id)
+    elif isinstance(e, BotocoreClientError):
+        _handle_dependency_botocore_error(e, task_id)
+    elif isinstance(e, ArrowException):
+        _handle_dependency_pyarrow_error(e, task_id)
+
```

Comment:
Are we simply ignoring the error if it doesn't fall into any of these categories? I think we should an `UnknownDeltaCATError` which indicates an immediate job failure.